cbor.js

```diff
@@ -111,42 +271,21 @@ function encode(value) {
 
     switch (typeof value) {
       case "number":
+          if (isNaN(value))
+            return writeFloat16(value);
         if (Math.floor(value) === value) {
           if (0 <= value && value <= POW_2_53)
             return writeTypeAndLength(0, value);
           if (-POW_2_53 <= value && value < 0)
             return writeTypeAndLength(1, -(value + 1));
         }
-        writeUint8(0xfb);
-        return writeFloat64(value);
 
-      case "string":
-        var utf8data = [];
-        for (i = 0; i < value.length; ++i) {
-          var charCode = value.charCodeAt(i);
-          if (charCode < 0x80) {
-            utf8data.push(charCode);
-          } else if (charCode < 0x800) {
-            utf8data.push(0xc0 | charCode >> 6);
-            utf8data.push(0x80 | charCode & 0x3f);
-          } else if (charCode < 0xd800) {
-            utf8data.push(0xe0 | charCode >> 12);
-            utf8data.push(0x80 | (charCode >> 6)  & 0x3f);
-            utf8data.push(0x80 | charCode & 0x3f);
-          } else {
-            charCode = (charCode & 0x3ff) << 10;
-            charCode |= value.charCodeAt(++i) & 0x3ff;
-            charCode += 0x10000;
-
-            utf8data.push(0xf0 | charCode >> 18);
-            utf8data.push(0x80 | (charCode >> 12)  & 0x3f);
-            utf8data.push(0x80 | (charCode >> 6)  & 0x3f);
-            utf8data.push(0x80 | charCode & 0x3f);
-          }
-        }
+        //writeUint8(0xfb);
+        //return writeFloat64(value);
+        return writeFloat(value);
```

Comment:
writes 16, 32, or 64 values as long as they can encode and decode exactly as that value